Sanne/eindopdracht.js

```diff
@@ -1,56 +1,109 @@
-//opdracht 1: maak een functie om de volledige URL op te bouwen
-// goede antwoord:
+// Assignment 1 - Create an API-key on openweathermaps.com using a browser, and store this in a variable.
+// Correct answer:
+let apiKey = "c7d6b1959579335813b946da059d652b";
+
+//Assignment 2 - Figure out how the OpenWeatherMaps API-works, then build a function which takes two arguements: city and apiKey.
+//The function will then return the full URL.
+//Verify that the URL is working through your browser. The outcome will be an object formatted in JSON.
+//Correct answer:
 let urlBuilder = function (city, apiKey) {
-	let urlBase = "http://api.openweathermap.org/data/2.5/weather?q=";
+    const urlBase = "http://api.openweathermap.org/data/2.5/weather?q=";
     return urlBase + city + "&APPID=" + apiKey;
-}
+};
+// verify it's working by calling the function and logging the output to the console.
+// correct answer
+var urlAmsterdam = urlBuilder("Amsterdam", apiKey);
+console.log(urlAmsterdam);
 
-//opdracht 2: maak de async await fuctie af op de API aan te roepen (invulopdracht)
+// Assignment 3 - Finish the function below to create an async await function. The purpose of this function is to make sure the API call waits till the Javascript page has loaded as well.
 //[invullen] [invullen] apiCall(url) {
-  //const response = [invullen] fetch(url);
-  //const json = [innvullen](response.json())
-
-    //return json;
+//const response = [invullen] [invullen](url);
+//const json = [innvullen]([invullen].json())
+//return json;
 //}
 
-//goede antwoord:
+//Correct answer:
 async function apiCall(url) {
   const response = await fetch(url);
   const json = await(response.json());
-
-    return json;
+  return json;
 }

```

Comment:
Dit is een prima functie, omdat hij:
- duidelijk en kort is
- 1 ding doet
- een promise retourneert (en dat is altijd goed als je asynchrone dingen doet :-P)

Maar de volgende stap zou dan kunnen zijn, dat je een nieuwe functie schrijft om de max_temp en min_temp uit de json te halen; in plaats van het in dezelfde functie te stoppen.

Je code wordt dan zoiets:
```apiCall(url).then( json => {
  return { temp_min: json.main.temp_min, temp_max: json.main.temp_max };
});```

Je volgende stap (omzetten naar float bijvoorbeeld) zou je dan ook weer in een `.then` hierachter kunnen zetten.